aws/resource_aws_instance.go

```diff
@@ -1011,6 +1010,58 @@ func resourceAwsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("root_block_device") {
+		if rbd, ok := d.GetOk("root_block_device"); ok {
+			rbdL := rbd.([]interface{})
+			if len(rbdL) > 1 {
+				return fmt.Errorf("Cannot specify more than one root_block_device.")
+			}
+			for _, r := range rbdL {
+				bd := r.(map[string]interface{})
+				if volumeSize, ok := bd["volume_size"].(int); ok && volumeSize != 0 {
+					volumeIds, err := getAwsInstanceVolumeIds(conn, d)
+					if err != nil {
+						return fmt.Errorf("Error retrieving volumes: %s", err)
+					}
+
+					volResp, err := conn.DescribeVolumes(&ec2.DescribeVolumesInput{
+						VolumeIds: volumeIds,
+					})
+					if err != nil {
+						return err
+					}
+
+					if len(volResp.Volumes) < 1 {
+						return fmt.Errorf("Cannot fetch volume info")
```

Comment:
Do you mind making the error message a bit more helpful here? e.g. `"No volumes found for %s", d.Id()`